sic_io/src/load.rs

```diff
@@ -0,0 +1,321 @@
+use std::error::Error;
+use std::fs::File;
+use std::io::{BufReader, Read};
+use std::path::Path;
+
+use sic_core::image;
+use sic_core::image::AnimationDecoder;
+
+/// Load an image using a reader.
+/// All images are currently loaded from memory.
+pub fn load_image<R: Read>(
+    reader: &mut R,
```

Comment:
Just came across a newly introduced Reader for the image crate: https://github.com/image-rs/image/commit/ac93e755b51def00c93e7fadd33a5e5593afa1e2#diff-c42f25b7f2392d73d32de4183d660a9aR34

It is similar to the current implementation, so perhaps we can replace our implementation and use the Reader from the image crate on its next release.